app/controllers/your_representatives_controller.rb

```diff
@@ -1,5 +1,22 @@
 class YourRepresentativesController < ApplicationController
   def edit
-    head :ok
+    @your_representative = YourRepresentative.new
+  end
+
+  def update
+    @your_representative = YourRepresentative.new(your_representative_params)
+    if @your_representative.valid?
+      if @your_representative.have_representative
```

Comment:
Maybe extract this into separate method ? with the redirect_to being called in here, just the url being returned from this separate method

Reply:
Great point, thanks. Refactored in de43ff2